libs/types/src/solana/mod.rs

```diff
@@ -1,11 +1,76 @@
-use crate::MultiRpcResult;
 use candid::CandidType;
 use serde::{Deserialize, Serialize};
 use std::fmt::Debug;
 
 /// A Solana [slot](https://solana.com/docs/references/terminology#slot).
 pub type Slot = u64;
 
+/// A Solana base58-encoded [transaction ID](https://solana.com/docs/references/terminology#transaction-id).
+pub type TransactionId = String;
+
+/// The parameters for a Solana [`sendTransaction`](https://solana.com/docs/rpc/http/sendtransaction) RPC method call.
+#[derive(Debug, Clone, Deserialize, Serialize, CandidType)]
+pub struct SendTransactionParams {
+    /// Fully-signed transaction, as encoded string.
+    pub transaction: String,
+    /// Encoding format for the transaction.
+    pub encoding: Option<SendTransactionEncoding>,
+    /// When true, skip the preflight transaction checks. Default: false.
+    #[serde(rename = "skipPreflight")]
+    pub skip_preflight: Option<bool>,
+    /// Commitment level to use for preflight. See Configuring State Commitment. Default finalized.
+    #[serde(rename = "preflightCommitment")]
+    pub preflight_commitment: Option<String>,
+    /// Maximum number of times for the RPC node to retry sending the transaction to the leader.
+    /// If this parameter not provided, the RPC node will retry the transaction until it is
+    /// finalized or until the blockhash expires.
+    #[serde(rename = "maxRetries")]
+    pub max_retries: Option<u32>,
+    /// Set the minimum slot at which to perform preflight transaction checks
+    #[serde(rename = "minContextSlot")]
+    pub min_context_slot: Option<u64>,
+}
+
+impl SendTransactionParams {
+    /// Returns `true` if all of the optional config parameters are `None` and `false` otherwise.
+    pub fn is_default_config(&self) -> bool {
+        self.encoding.is_none()
+            && self.skip_preflight.is_none()
+            && self.preflight_commitment.is_none()
+            && self.max_retries.is_none()
+            && self.min_context_slot.is_none()
+    }
```

Comment:
nit: I would use the destructuring pattern to catch the addition of any optional field
```suggestion
    pub fn is_default_config(&self) -> bool {
        let SendTransactionParams {
            transaction: _,
            encoding,
            skip_preflight,
            preflight_commitment,
            max_retries,
            min_context_slot,
        } = &self;
        encoding.is_none()
            && skip_preflight.is_none()
            && preflight_commitment.is_none()
            && max_retries.is_none()
            && min_context_slot.is_none()
    }
```

Reply:
Good idea! Done.